### PR TITLE
Add additional log message for unexpected response code

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Adds additional log message if we get a non 200 or 401 response to a data query

## Why are we making these changes?
To improve debugging for requests that don't return any data.

### CODE REVIEW

File: cli/cook/http.py [hunk 1]
```diff
@@ -103,6 +103,8 @@ def make_data_request(cluster, make_request_fn):
             return resp.json()
         elif resp.status_code == 401:
             print_error(f'Authentication failed on {cluster["name"]} ({cluster["url"]}).')
+        else:
+            logging.warn(f'Unexpexted response code {resp.status_code} for data request. Response body: {resp.text}')
```
Comment (dposada) [inline cli/cook/http.py]:
```suggestion
            logging.warn(f'Unexpected response code {resp.status_code} for data request. Response body: {resp.text}')
```

Comment (pschorf) [reply cli/cook/http.py]: Fixed

File: cli/cook/http.py [hunk 2]
```diff
@@ -103,6 +103,13 @@ def make_data_request(cluster, make_request_fn):
             return resp.json()
         elif resp.status_code == 401:
             print_error(f'Authentication failed on {cluster["name"]} ({cluster["url"]}).')
+            return []
+        elif resp.status_code == 500:
+            print_error(f'Encountered server error while querying {cluser["name"]}')
```
Comment (dposada) [inline cli/cook/http.py]:
```suggestion
            print_error(f'Encountered server error while querying {cluser["name"]}.')
```

**Comment by dposada** [comment]
Will merge when build is green

**Pull request merged**